### PR TITLE
chore(ci): test core_crypto with avx512

### PR DESCRIPTION
**Pull request by soonum**

### CODE REVIEW

File: .github/workflows/cargo_build.yml [hunk 1]
```diff
@@ -31,6 +31,10 @@ jobs:
         run: |
           make build_core
 
+      - name: Build Release core (with AVX512)
+        run: |
+          AVX512_SUPPORT=ON make build_core
```
Comment (IceTDrinker) [inline .github/workflows/cargo_build.yml:36]: this won't play nice with windows I believe

File: Makefile [hunk 2]
```diff
@@ -174,7 +174,7 @@ build_node_js_api: install_rs_build_toolchain
 .PHONY: test_core_crypto # Run the tests of the core_crypto module including experimental ones
 test_core_crypto: install_rs_build_toolchain
 	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) test --release \
-		--features=$(TARGET_ARCH_FEATURE),experimental -p tfhe -- core_crypto::
+		--features=$(TARGET_ARCH_FEATURE),experimental,$(AVX512_FEATURE) -p tfhe -- core_crypto::
```
Comment (IceTDrinker) [inline Makefile:177]: I think the build_core target update is missing ?

Comment (IceTDrinker) [reply Makefile:177]: also here I would like to have both tests, once without avx512 and once with it as it changes the code being executed

Comment (IceTDrinker) [reply Makefile:177]: depending on whether AVX512 support was ON or not

Comment (soonum) [reply Makefile:177]: I don't quite get it about the both tests thing. You're not happy by having two workflow jobs for testing ?

Comment (IceTDrinker) [reply Makefile:177]: No I mean that we want to run the test once in this target without avx512 so the existing command is ok and then we run conditionally a second command with avx512 support on and nightly toolchain

As both code paths need to be correct

File: Makefile [hunk 3]
```diff
@@ -126,11 +136,13 @@ gen_key_cache: install_rs_build_toolchain
 		--features=$(TARGET_ARCH_FEATURE),shortint,internal-keycache -p tfhe
 
 .PHONY: build_core # Build core_crypto with and without experimental features
-build_core: install_rs_build_toolchain
+build_core: install_rs_build_toolchain install_rs_build_toolchain_nightly
```
Comment (IceTDrinker) [inline Makefile:139]: I think you could have used the nighlty check toolchain instead ? Or we can rename the check toolchain to nightly toolchain

File: Makefile [hunk 4]
```diff
@@ -126,11 +136,13 @@ gen_key_cache: install_rs_build_toolchain
...
 	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) build --release \
 		--features=$(TARGET_ARCH_FEATURE) -p tfhe
-	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) build --release \
-		--features=$(TARGET_ARCH_FEATURE),experimental -p tfhe
+	@if [[ "$(AVX512_SUPPORT)" == "ON" ]]; then \
+		RUSTFLAGS="$(RUSTFLAGS)" cargo +nightly build --release \
+			--features=$(TARGET_ARCH_FEATURE),experimental,$(AVX512_FEATURE) -p tfhe; \
+	fi
```
Comment (IceTDrinker) [inline Makefile:145]: if you look closely we build once with experimental, here if we don't have the avx512 we won't build with experimental anymore

Comment (soonum) [reply Makefile:145]: Copy paste mistake

**Review by IceTDrinker — APPROVED** [review]
Thanks for the PR !

**Pull request merged**